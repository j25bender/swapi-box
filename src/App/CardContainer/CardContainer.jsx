import React from 'react';
import Card from '../Card/Card';
import './CardContainer.css'

const CardContainer = (props) => {
  return (
    <div className="cardContainer">
      {
        props.fetch.map( (data, index) => <Card key={index} data={data} selectCard={props.selectCard} /> )
      }
    </div>
  );
}

export default CardContainer