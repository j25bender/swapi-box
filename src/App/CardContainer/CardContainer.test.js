import React from 'react'
import Enzyme, {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import CardContainer from './CardContainer'

Enzyme.configure({ adapter: new Adapter() })

describe('CardContainer', () => {
    const mockselectCard = jest.fn()

    it('matches the snapshot', () => {
        const peopleState = [
            "favorite": false, 
            "info": {
                "Homeworld": "Naboo", 
                "Name": "Luke Skywalker", 
                "Population": 4500000000, 
                "Species": "Human"}
            ]

        const wrapper = shallow(<CardContainer name="people" selectCard={mockselectCard} fetch={ peopleState } />)
        
        expect(wrapper).toMatchSnapshot()
    })
})