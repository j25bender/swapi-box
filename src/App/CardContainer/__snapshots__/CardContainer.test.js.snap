// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    fetch={
        Array [
            "favorite",
            "info",
          ]
    }
    name="people"
    selectCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="cardContainer"
>
        <Card
                data="favorite"
                selectCard={[Function]}
        />
        <Card
                data="info"
                selectCard={[Function]}
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          Array [
            <Card
              data="favorite"
              selectCard={[Function]}
/>,
            <Card
              data="info"
              selectCard={[Function]}
/>,
          ],
        ],
        "className": "cardContainer",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "data": "favorite",
            "selectCard": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "data": "info",
            "selectCard": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="cardContainer"
>
          <Card
                    data="favorite"
                    selectCard={[Function]}
          />
          <Card
                    data="info"
                    selectCard={[Function]}
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            Array [
              <Card
                data="favorite"
                selectCard={[Function]}
/>,
              <Card
                data="info"
                selectCard={[Function]}
/>,
            ],
          ],
          "className": "cardContainer",
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "data": "favorite",
              "selectCard": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "data": "info",
              "selectCard": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
